testing/openjfx12/APKBUILD

```diff
@@ -0,0 +1,241 @@
+COMPILE_MEDIA=true
+# non standard, alpine patched properties
+ANT_HOME=$_ant_home
+EXTRA_CXXFLAGS=$CXXFLAGS
+EXTRA_LDFLAGS=$LDFLAGS
+PKG_CONFIG_PATH=$PKG_CONFIG_PATH
+PKG_CONFIG_SYSROOT_DIR=$PKG_CONFIG_SYSROOT_DIR
+CC=${CC:-gcc}
+CXX=${CXX:-g++}
+AR=${AR:-ar}
+EOF
+
+	# use system provided library for icu
+	rm -rf modules/javafx.web/src/main/native/Source/ThirdParty/icu
+
+	# make gradle wrapper executable
+	chmod +x ./gradlew
+}
+
+build() {
+	./gradlew publicExportsStandaloneLinux
+}
+
+check() {
+	./gradlew check -x :web:test
+}
+
+package() {
+	mkdir -p "$pkgdir/$_java_home/apps"
+	cp -r build/artifacts/javafx-sdk-12*/*                         "$pkgdir/$_java_home"
+	cp -r build/artifacts/javafx-jmods-12*/                        "$pkgdir/$_java_home/jmods"
+	cp -r build/artifacts/javafx-docs-12*/                         "$pkgdir/$_java_home/api"
+}
+
+_jmods() {
+	pkgdesc="$pkgdesc (jmods)"
+	depends=""
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot"
+	mv "$_fromroot/jmods" "$_toroot"
+}
+
+_src() {
+	pkgdesc="$pkgdesc (src)"
+	depends=""
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot/lib"
+	mv "$_fromroot/lib/src.zip" "$_toroot/lib/src.zip"
+}
+
+_doc() {
+	pkgdesc="$pkgdesc (Documentation)"
+	depends=""
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot"
+	mv "$_fromroot/api" "$_toroot"
+}
+
+_gtk2() {
+	pkgdesc="$pkgdesc (GTK 2)"
+	depends="$pkgname-base"
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot/lib"
+	mv "$_fromroot/lib/libglassgtk2.so"   "$_toroot/lib"
+}
+
+_gtk3() {
+	pkgdesc="$pkgdesc (GTK 3)"
+	depends="$pkgname-base"
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot/lib"
+	mv "$_fromroot/lib/libglassgtk3.so"   "$_toroot/lib"
+}
+
+_web() {
+	pkgdesc="$pkgdesc (web)"
+	depends="$pkgname-media"
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot/lib"
+	mkdir -p "$_toroot/jmods"
+	mkdir -p "$_toroot/legal"
+	mv \
+		"$_fromroot"/lib/libjfxwebkit.so \
+		"$_fromroot"/lib/javafx.web.jar \
+		"$_toroot/lib"
+	mv \
+		"$_fromroot"/jmods/javafx.web.jmod \
+		"$_toroot/jmods"
+	mv \
+		"$_fromroot/legal/javafx.web" \
+		"$_toroot/legal"
+}
+
+_media() {
+	pkgdesc="$pkgdesc (media)"
+	depends="$pkgname-libs"
+	_fromroot="$pkgdir/$_java_home"
+	_toroot="$subpkgdir/$_java_home"
+
+	mkdir -p "$_toroot/lib"
+	mkdir -p "$_toroot/jmods"
+	mkdir -p "$_toroot/legal"
+
+	mv \
+		"$_fromroot"/lib/libavplugin.so \
+		"$_fromroot"/lib/libgstreamer-lite.so \
+		"$_fromroot"/lib/libjfxmedia.so \
+		"$_fromroot"/lib/libfxplugins.so \
+		"$_fromroot"/lib/javafx.media.jar \
+		"$_toroot/lib"
+	mv \
+		"$_fromroot"/jmods/javafx.media.jmod \
+		"$_toroot/jmods"
+	mv \
+		"$_fromroot/legal/javafx.media" \
+		"$_toroot/legal/"
+}
+
+_libs() {
+	pkgdesc="$pkgdesc (libs)"
+	depends=""
```

Comment:
Nitty, but you can remove these empty variables

Reply:
The "root" package has a depends, so i wanted to make sure this subpackage does not inherit the root one. Can I remove this empty assignment?

Reply:
Hm, I'm like 99% sure that the subpkg won't inherit the deps, but let me check